### PR TITLE
[WIP] New scheduler placement groups 

### PR DESCRIPTION
**Pull request by wuisawesome**
## Why are these changes needed?

Placement group implementation for the new scheduler. This also means supporting custom dynamic resources in the cluster resource scheduler, but we aren't exposing this to the user for now. 

## Related issue number

#9801

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by ericl** [comment]
cc @raulchen @rkooo567 @clay4444

File: src/ray/raylet/scheduling/cluster_task_manager.cc [hunk 1]
```diff
@@ -329,6 +330,50 @@ void ClusterTaskManager::Heartbeat(bool light_heartbeat_enabled,
...
 
+bool ClusterTaskManager::PreparePGBundle(BundleSpecification &bundle_spec) {
+  if (pg_bundles_.count(bundle_spec.BundleId()) > 0) {
+    // Duplicate prepare request.
+    return true;
+  }
+
+  std::shared_ptr<TaskResourceInstances> resource_instances =
+      std::make_shared<TaskResourceInstances>();
+  bool allocated = cluster_resource_scheduler_->AllocateLocalTaskResources(
```
Comment (clay4444) [inline src/ray/raylet/scheduling/cluster_task_manager.cc:341]: we use `cluster_resource_scheduler_ ` instead of `scheduling_resource` in new scheduler?

Comment (wuisawesome) [reply src/ray/raylet/scheduling/cluster_task_manager.cc:341]: Yes, the state of the local node (resource instances) and the node's "view" of the cluster is managed by the cluster resource scheduler.

**Comment by rkooo567** [comment]
@clay4444 @wuisawesome Can I close this? I feel like this PR is what you guys need right? https://github.com/ray-project/ray/pull/12518

**Comment by wuisawesome** [comment]
Can we leave this open for a bit? I think the changes to `cluster_task_manager.cc` here are still useful.

**Comment by wuisawesome** [comment]
btw @clay4444 you should be able to just resolve these conflicts by just checking the files out of `master`, the conflicts are because the changes were pulled out into #12518.

**Comment by clay4444** [comment]
> btw @clay4444 you should be able to just resolve these conflicts by just checking the files out of `master`, the conflicts are because the changes were pulled out into #12518.

yeah, I get that.

**Comment by wuisawesome** [comment]
Closing now that #12910 exists

**Pull request closed without merging**